File: python/leetcode/349-intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        hash = {}
        for i in range(len(nums1)):
            if nums1[i] not in hash:
                hash[nums1[i]] = 1
        
        s = set()
        for j in range(len(nums2)):
            if nums2[j] in hash:
                s.add(nums2[j])
        
        return list(s)
            
        
    def intersectionFast(self, nums1: List[int], nums2: List[int]) -> List[int]:
        s1, s2 = set(nums1), set(nums2)
        return list(s1.intersection(s2))
        

File: c#/leetcode/349-intersection-of-two-arrays.cs
public class Solution {
    public int[] Intersection(int[] nums1, int[] nums2) {
        HashSet<int> s1 = new HashSet<int>(nums1);
        HashSet<int> s2 = new HashSet<int>(nums2);
        s1.IntersectWith(s2);
        return s1.ToArray();
    }
}